dask_expr/_util.py

```diff
@@ -184,6 +186,28 @@ def _maybe_from_pandas(dfs):
     return dfs
 
 
+def _get_shuffle_preferring_order(shuffle):
+    if shuffle is not None:
+        return shuffle
+
+    if get_default_shuffle_method() == "disk":
+        return "tasks"
+
+    return get_default_shuffle_method()
```

Comment:
Why are we diverging from the "usual" defaults here?

nit:
```suggestion
    shuffle = get_default_shuffle_method()
    if shuffle == "disk":
        return "tasks"

    return shuffle
```